java.js

```diff
@@ -53,27 +54,29 @@ export default class Java extends Tool {
             if (!match) continue
             found[match[1]] = match[2]
         }
-        this.debug(`Found .sdkman entries ${JSON.stringify(found)}`)
+        this.debug(`Found .sdkmanrc entries ${JSON.stringify(found)}`)
         return found["java"]
     }
 
-    // stripJavaVersionSuffix returns the normalized version without suffix strings
-    stripJavaVersionSuffix(versionIdentifier, suffixIdentifier) {
-        const suffixStartIndex = versionIdentifier.indexOf(suffixIdentifier)
-        return suffixStartIndex >= 0
-            ? versionIdentifier.substring(0, suffixStartIndex)
-            : versionIdentifier
+    // parseJavaVersionString specially handles version string extraction
+    // because we have to map strings like 1.8.0_282 to 8.0.282 for the actual
+    // SemVer comparison
+    parseJavaVersionString(expected, version) {
+        // Default case for 11.x or 17.x it should match and we're ok
+        let versions = findVersions(version, { loose: true })
+        if (versions && versions[0] == expected) return versions
+
+        // Handle bullshit case
```

Comment:
Would be good to tell the reader why these four lines of code are necessary.

Reply:
Ugh, fine, I _guess_. :joy:

Reply:
Ugh fine, I _guess_. 😂